routes/users.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const jwt = require('jsonwebtoken');
+const { JWT_SECRET = 'prestons-secret-isnt-secret' } = process.env ; 
+const SALT_COUNT = 10;
+const { createUser, getUser, getUserByUserName } = require('../db/users');
+const { getOrdersByUser } = require('../db/orders');
+const usersRouter = express.Router();
+
+
+ function requireUser(req, res, next) {
+     if(!req.user) {
+         next({
+             name: 'MissingUserError',
+             message: 'You must be logged in to perform this action'
+         });
+     } 
+         next();
+ };
+
+
+usersRouter.post('/register', async (req, res, next) => {
+    const { imageURL, firstName, lastName, email, username, password } = req.body;
+    try{
+        const _user = await getUserByUserName(username);
+        if (_user) {
+            res.send({
+                name: 'UserExistsError',
+                message: 'A user by that username already exists'
+            });
+            return
+        } else if (password.length < 8) {
+            res.send({
+                name: 'PasswordLengthError',
+                message: 'Password Too Short!'
+            });
+            return
+        } else {
+            const user = await createUser({ firstName, lastName, email, imageURL, username, password });
+            
+            const token = jwt.sign({ 
+                id: user.id, 
+                username
+              }, JWT_SECRET, {
+                expiresIn: '1w'
+              })
+            console.log(user)
+            res.send({
+                user,
+                message: "Thank you for signing up",
+                token
+            });
+        }
+    } catch (error) {
+        next(error);
+    }    
+});
+
+usersRouter.post('/login', async (req, res, next) => {
+    try {
+        const { username, password } = req.body;
+        if (!username || !password) {
+            next({
+                name: "MissingCredentialsError",
+                message: "Please supply both a username and password"
+            });
+        } else {
+
+        const user = await getUser({username, password});
+        req.user = user;
+        if (user) { //we have a valid user, successful login
+            const token = jwt.sign({
+                id: user.id,
+                username: user.username
+                }, process.env.JWT_SECRET, {
```

Comment:
Should just use `JWT_SECRET` since we already pulled in the process env variable above (and we set the default above too, in case env isn't defined)